tests/testsuite/init.rs

```diff
@@ -598,3 +598,68 @@ mod tests {
 "#
     );
 }
+
+#[cargo_test]
+fn creates_binary_when_instructed_and_has_lib_file_no_warning() {
+    let path = paths::root().join("foo");
+    fs::create_dir(&path).unwrap();
+    fs::write(path.join("foo.rs"), "fn not_main() {}").unwrap();
+    cargo_process("init --bin")
+        .cwd(&path)
+        .with_stderr(
+            "[WARNING] file 'foo.rs' seems to be a library file\n[CREATED] binary (application) package"
+        )
+        .run();
+
+    let cargo_toml = fs::read_to_string(path.join("Cargo.toml")).unwrap();
+    assert!(cargo_toml.contains("[[bin]]"));
+    assert!(!cargo_toml.contains("[lib]"));
+}
+
+#[cargo_test]
+fn creates_library_when_instructed_and_has_bin_file() {
+    let path = paths::root().join("foo");
+    fs::create_dir(&path).unwrap();
+    fs::write(path.join("foo.rs"), "fn main() {}").unwrap();
+    cargo_process("init --lib")
+        .cwd(&path)
+        .with_stderr(
+            "[WARNING] file 'foo.rs' seems to be a binary (application) file\n[CREATED] library package"
```

Comment:
Prefer this multi-line style for being consistent with other tests

```suggestion
            "\
[WARNING] file 'foo.rs' seems to be a binary (application) file
[CREATED] library package
"
```